### PR TITLE
Batch processing

### PR DESCRIPTION
**Pull request by epureanudiana**
process batch of images from folder, save positive and negative detection results in another folder.

### CODE REVIEW

File: predictions.py [hunk 1]
```diff
@@ -109,6 +117,8 @@ def single_instance_prediction(
     predictor = DefaultPredictor(cfg)
     outputs = predictor(im)
 
+    print(outputs["instances"])
```
Comment (thomasbrockmeier-ams) [inline predictions.py:120]: Is this a debugging statement, or did you leave it here on purpose?

Looking at just the diffs, it seems that some of the import statements at the top of this file are not used.

File: utils.py [hunk 2]
```diff
@@ -379,18 +382,54 @@ def handle_hyperparameters(config: Union[str, Path]) -> int:
...
-        )
+        try:
+            with open(ann_path) as f:
+                _ = json.load(f)
+                register_coco_instances(f"{expCfg.dataset_name}_{expCfg.subset}", {}, ann_path, image_root=f"{expCfg.data_folder}")
+        except FileNotFoundError:
+            if expCfg.subset == "test":
+                add_images_to_coco(image_dir=f"{expCfg.data_folder}/{expCfg.subset}",
+                                   coco_filename=f"{expCfg.data_folder}/{expCfg.subset}/containers-annotated-COCO-{expCfg.subset}.json")
+                time.sleep(5)
```
Comment (thomasbrockmeier-ams) [inline utils.py:422]: This should go, right?

**Review by thomasbrockmeier-ams — COMMENTED** [review]
What's going on with the sleep statement? Looks good otherwise!

File: inference.py [hunk 3]
```diff
@@ -95,6 +68,11 @@ def evaluate_model(flags: argparse.Namespace, expCfg: ExperimentConfig) -> None:
     constructor from another method i.e. _test_loader_from_config
     """
 
+    ws = Workspace.from_config()
```
Comment (laurenssam) [inline inference.py:71]: Does this work when we running from the production pipeline. Is the config for our ml studio or is this a more personal check?

Comment (laurenssam) [reply inference.py:71]: + we should think about how we get this config file safely into the environment!

File: inference.py [hunk 4]
```diff
@@ -95,6 +68,11 @@ def evaluate_model(flags: argparse.Namespace, expCfg: ExperimentConfig) -> None:
...
+
+    _ = Model.get_model_path(
+        model_name=f"{flags.name}", version=int(flags.version), _workspace=ws
```
Comment (laurenssam) [inline inference.py:74]: Is the default here to get the latest one?

File: inference.py (hunk 3 above)
Comment (laurenssam) [inline inference.py:71]: Does this work in a container? And if so, we should think about how we get the config file safely into the production environment

File: inference.py [hunk 5]
```diff
@@ -120,10 +98,13 @@ def evaluate_model(flags: argparse.Namespace, expCfg: ExperimentConfig) -> None:
...
 
     flags = arg_parser()
+
     experimentConfig = ExperimentConfig(
         dataset_name=flags.dataset_name,
         subset=flags.subset,
         data_format=flags.data_format,
         data_folder=flags.data_folder,
     )
+
```
Comment (laurenssam) [inline inference.py:108]: So this function creates a coco style prediction file, which goes into the postprocessing right?

Comment (epureanudiana) [reply inference.py:108]: yes, correct. it creates 2 output files: the one with positive results and the one with negative results.

File: predictions.py (hunk 1 above)
Comment (epureanudiana) [reply predictions.py]: thanks, I deleted the redundant lines.

File: utils.py (hunk 2 above)
Comment (epureanudiana) [reply utils.py]: yes!

File: inference.py (hunk 4 above)
Comment (epureanudiana) [reply inference.py]: the default was version 1, now I changed it to download the latest version.

**Pull request merged**